operator/v1/types_console.go

```diff
@@ -143,8 +143,140 @@ type Capability struct {
+	// +required
+	From SourceType `json:"from"`
+
+	// configMap specifies the ConfigMap sourcing details such as the name of the ConfigMap and the key for the file.
+	// The ConfigMap must exist in the openshift-config namespace.
+	// Required when from is "ConfigMap", and forbidden otherwise.
+	// +optional
+	ConfigMap *ConfigMapFileReference `json:"configMap"`
+}
+
+// Theme defines a theme mode for the console UI.
+type Theme struct {
+	// mode is used to specify what theme mode a logo will apply to in the console UI.
+	// mode is a required field that allows values of Dark and Light.
+	// When set to Dark, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Dark mode.
+	// When set to Light, the logo file referenced in the 'file' field will be used when an end-user of the console UI enables the Light mode.
+	// +required
+	Mode ThemeMode `json:"mode"`
+
+	// source is used by the console to locate the specified file containing a custom logo.
+	// source is a required field that references a ConfigMap name and key that contains the custom logo file in the openshift-config namespace.
+	// You can create it with a command like:
+	// - 'oc create configmap custom-logos-config --namespace=openshift-config --from-file=/path/to/file'
+	// The ConfigMap key must include the file extension so that the console serves the file with the correct MIME type.
+	// The recommended file format for the Masthead and Favicon logos is SVG, but other file formats are allowed if supported by the browser.
+	// The logo image size must be less than 1 MB due to constraints on the ConfigMap size.
+	// For more information, see the documentation: https://docs.redhat.com/en/documentation/openshift_container_platform/4.19/html/web_console/customizing-web-console#customizing-web-console
+	// +required
+	Source FileReferenceSource `json:"source"`
+}
+
+// Logo defines a configuration based on theme modes for the console UI logo.
+type Logo struct {
+	// type specifies the type of the logo for the console UI. It determines whether the logo is for the masthead or favicon.
+	// type is a required field that allows values of Masthead and Favicon.
+	// When set to "Masthead", the logo will be used in the masthead and about modal of the console UI.
+	// When set to "Favicon", the logo will be used as the favicon of the console UI.
+	// +required
+	Type LogoType `json:"type"`
+
+	// themes specifies the themes for the console UI logo.
+	// themes is a required field that allows a list of themes. Each item in the themes list must have a unique mode and a source field.
+	// Each mode determines whether the logo is for the dark or light mode of the console UI.
+	// There must be at least one entry and no more than 2 entries.
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=map
+	// +listMapKey=mode
+	// +required
+	Themes []Theme `json:"themes"`
```

Comment:
Your godoc should explain that the default openshift logo will be displayed for any theme that is not specified